tasmota/xnrg_12_solaxX1.ino

```diff
@@ -131,15 +107,16 @@ uint8_t functionCode[1] = {0x00};
 uint8_t dataLength[1] = {0x00};
 uint8_t data[16] = {0};
 
+TasmotaSerial *solaxX1Serial;
 uint8_t message[30];
+bool AddressAssigned = true;
+uint8_t solaxX1_send_retry = 20;
+uint8_t solaxX1_queryData_count = 0;
+uint8_t solaxX1_QueryID_count = 240;
+uint8_t solaxX1SerialNumber[16] = {0x6e, 0x2f, 0x61}; // "n/a"

```

Comment:
Pls move all these global variables into the ``solaxX1`` struct to fix future global variable naming issues.

Reply:
@arendst Thanks for your advice. As I am working on extensions to request more information from the inverter I will implement your requested changes in my current work. So you can merge this PR and I will start a new PR in the next few days. Otherwise I can publish a new commit to this PR with all the changes and additions.

Reply:
@arendst The work is done and I commited the changes.